lookout/core/metrics.py

```diff
@@ -1,12 +1,197 @@
+        PROMETHEUS_SERVER.start_http_server()
+    elif not PROMETHEUS_SERVER.is_running:
+        PROMETHEUS_SERVER.start_http_server()
+
+
+def submit_event(key: str, value: Union[int, float, bool], labelnames: str = ""):
+    """Register an event by a key and with a numeric value.
+
+    If the key does not exist, it creates a new Prometheus Metric.
 
     :param key: Identifier of the event.
-    :param value: Value of the event.
+    :param value: Value of the event. It will convert cast variables to int.
+    :param labelnames: Additional description of the event. Only used when creating a new event.
     :return: None
     """
-    pass
+    start_prometheus()
+    if PROMETHEUS_SERVER is not None:
+        PROMETHEUS_SERVER.submit_event(key=key, value=value, labelnames=labelnames)
+    else:
+        raise ValueError(
+            "PrometheusServer is not started yet, please call start_prometheus() first.",
+        )
+
+
+class PrometheusServer:
+    """Manage the streaming process for different metrics."""
+
+    def __init__(self, port=PROMETHEUS_PORT, addr=PROMETHEUS_HOST):
+        """
+         Manage the streaming process for different metrics.
+
+        :param port:
+        :param addr:
+        """
+        self._is_running = False
+        self._port = port
+        self._addr = addr
+        self._metrics = {}
+        self._invalid_punctuation = "".join(set(string.punctuation) - set(["_"]))
+
+    @property
+    def port(self) -> int:
+        """Return the port where the server is running."""
+        return self._port
+
+    @property
+    def addr(self) -> str:
+        """Return the address where the server is running."""
+        return self._addr
+
+    @property
+    def metrics(self) -> dict:
+        """Return the metrics stored in the server."""
+        return self._metrics
+
+    @property
+    def is_running(self) -> bool:
+        """Return True if the server is running."""
+        return self._is_running
+
+    def start_http_server(self):
+        """Start the HTTP server."""
+        start_http_server(port=self.port, addr=self.addr)
+        self._is_running = True
+
+    def create_new_metric(self, name: str, labelnames: str = "", *args, **kwargs):
+        """Create a new metric in case it does not previously exists.
+
+        :param name: It will be used as a key to identify the new metric.
+        :param labelnames: Additional description of the event. Only used when creating a new
+            event.
+        :param args: Additional args to initialize the target metric_type.
+        :param kwargs: Additional kwargs to initialize the target metric_type.
+        :return: None
+        """
+        self.metrics[name] = ConfidentCounter(name, labelnames, *args, **kwargs)
+
+    def _filter_metric_name(self, name: str):
+        name = name.replace(".", "_")
+        return name.translate(str.maketrans("", "", self._invalid_punctuation))
```

Comment:
Everything which is not `[a-zA-Z0-9_]+` is a ValueError - our rules

Reply:
Then in pretty much all the times that you call `submit_event` across the project you are doing it wrong. For example, take a look at:
 https://github.com/src-d/lookout-sdk-ml/blob/47f96120cb415dd15c98906ee161cd9085257d73/lookout/core/manager.py#L73

Reply:
Good catch, I missed the dot. `[a-zA-Z0-9_.]+`

Reply:
I think Prometheus supports namespaces (please check!) so we should not convert dots to underscores btw

Reply:
So should I remove the filtering of invalid names??

Reply:
https://prometheus.io/docs/practices/naming/

The also have the option to add an aditional labelname, which is already supported in `submit_event`